addons/mail/static/src/core/common/attachment_view.js

```diff
@@ -73,13 +71,25 @@ export class AttachmentView extends Component {
     }
 
     popoutAttachment() {
-        this.mailPopoutService.popout(this.__owl__.bdom.parentEl).focus();
+        class PopupWrapper extends Component {
+            static template = xml`<div class="o_attachment_preview" style="display: block;"><AttachmentView t-props="props"/> </div>`;
```

Comment:
```suggestion
            static template = xml`<div class="o_attachment_preview" style="display: block;"><t t-component="component" t-props="props"/> </div>`;
```
The service should be allowed to add any component as attachment